packages/docs/docs/core-api/data-schema.md

```diff
@@ -2,15 +2,100 @@
 title: Data Schema 
 ---
 
-User-defined data schemas can be used when interfacing with the exposed `uint253` attestation data. This tool encodes and decodes attestation data, consolidating a user-defined attestation changes. 
+The `DataSchema` class abstracts UniRep data into a JavaScript object. This class can be used to encode and decode attestation data, and build attestations that are ready to be submitted to the UniRep smart contract.
+
+```ts
+import { Attestation, DataSchema, SchemaField } from '@unirep/core'
+
+const schema: SchemaField[] = [
+    {
+        name: 'posRep',
+        type: 'uint64',
+        updateBy: 'sum',
+    },
+    {
+        name: 'negRep',
+        type: 'uint64',
+        updateBy: 'sum',
+    },
+    {
+        name: 'graffiti',
+        type: 'uint205', // replacement fields must be `uint205`
+        updateBy: 'replace',
+    },
+    {
+        name: 'postCount',
+        type: 'uint49',
+        updateBy: 'sum',
+    },
+    {
+        name: 'commentCount',
+        type: 'uint49',
+        updateBy: 'sum',
+    },
+    {
+        name: 'voteCount',
+        type: 'uint49',
+        updateBy: 'sum',
+    },
+]
+
+const d = new DataSchema(schema)
+
+// The schema is parsed in the `DataSchema` constructor, but can also be manually parsed
```

Comment:
I think this part is unnecessary, I imagined `parseSchema` being a private function

Reply:
Deleted